packages/repository/src/define-repository-class.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import assert from 'assert';
+import {PrototypeOf} from './common-types';
+import {Entity, Model} from './model';
+import {
+  DefaultCrudRepository,
+  DefaultKeyValueRepository,
+  juggler,
+  Repository,
+} from './repositories';
+
+/**
+ * Signature for a Repository class bound to a given model. The constructor
+ * accepts only the dataSource to use for persistence.
+ *
+ * `define*` functions return a class implementing this interface.
+ *
+ * @typeParam M - Model class
+ * @typeParam R - Repository class/interface
+ */
+export interface NamedRepositoryClass<
```

Comment:
The name of this class is a discussion point we haven't resolved yet. See https://github.com/strongloop/loopback-next/pull/4949#discussion_r400189603.

Cross-posting my latest comment:

> Maybe we should take another angle. My intention is to distinguish between a generic repository class that works for any Model/Entity and a repository class that's already specialized for (bound to) a given Model/Entity.
> 
> In that light, perhaps `BoundRepositoryClass` or `SpecializedRepositoryClass` would be better names?

Reply:
Maybe `TypedRepositoryClass`?

Reply:
I renamed it to `ModelRepositoryClass`. Let's just land it - no need to be perfect.